addon/components/data-filterer.js

```diff
@@ -0,0 +1,66 @@
+import Ember from 'ember';
+import layout from '../templates/data-filterer';
+import {diacriticsInsensitiveMatcher as defaultMatcher} from '../util/diactricts';
+const {computed, isEmpty, get} = Ember;
+
+function createFieldFilter(fieldName, filter) {
+  return {fieldName: fieldName, filter: filter};
+}
+
+function internalFilterer(data, filterFields) {
+  if (isEmpty(filterFields)) {
+    return data;
+  }
+
+  return data.filter(function (item) {
+    return itemFilterer(item, filterFields);
+  });
+}
+
+function itemFilterer(item, filterFields) {
+  let satisfiedFilters = filterFields.filter((filterField)=>{
+    let fieldName = get(filterField, 'fieldName');
+    let filter = get(filterField, 'filter');
+
+    return defaultMatcher(get(item, fieldName), filter);
+  });
+
+  return satisfiedFilters.length === filterFields.length;
+}
+
+export default Ember.Component.extend({
+  layout,
+  tagName:'',
+
+  init() {
+    this._super(...arguments);
+
+    if (isEmpty(this.get('fieldFilters'))) {
+      this.set('fieldFilters', Ember.A());
+    }
+  },
+
+  filteredData:computed('data', 'filterer', 'fieldFilters.[]', function() {
+    if (isEmpty(this.get('data'))) {
+      return [];
```

Comment:
Since we are using "ember-disable-prototype-extensions" in our addon, we may consider using Ember.A for this case.